cmd/podman/sign.go

```diff
@@ -95,10 +103,15 @@ func signCmd(c *cliconfig.SignValues) error {
 		if err != nil {
 			return errors.Wrapf(err, "error parsing image name")
 		}
-		rawSource, err := srcRef.NewImageSource(getContext(), runtime.SystemContext())
+		rawSource, err := srcRef.NewImageSource(getContext(), sc)
 		if err != nil {
 			return errors.Wrapf(err, "error getting image source")
 		}
+		defer func(rawSource types.ImageSource) {
```

Comment:
Do you actually need to pass in the rawSource? I think we can just use it within the defer functionality.

Reply:
I need to pass rawSource because this defer is in a for loop.
But I think I can directly using rawSource.Close() without using defer.